src/core/cff_parser.js

```diff
@@ -1413,7 +1394,27 @@ var CFFCompiler = (function CFFCompilerClosure() {
     compileNameIndex: function CFFCompiler_compileNameIndex(names) {
       var nameIndex = new CFFIndex();
       for (var i = 0, ii = names.length; i < ii; ++i) {
-        nameIndex.add(stringToBytes(names[i]));
+        var name = names[i];
+        var sanitizedName = '';
+        // OTS doesn't allow names to be over 127 characters.
+        var length = Math.min(name.length, 127);
+        for (var j = 0; j < length; j++) {
+          // OTS requires chars to be between a range and not certain other
+          // chars.
+          var char = name[j];
+          if (char < '!' || char > '~' || char === '[' || char === ']' ||
+              char === '(' || char === ')' || char === '{' || char === '}' ||
+              char === '<' || char === '>' || char === '/' || char === '%' ||
+              char === ' ') {
+            char = '_';
+          }
+          sanitizedName += char;
```

Comment:
Would it perhaps be more efficient to use an Array instead, and just insert the `char`s into it and then `join` at the end to get the string (in this case we even know the array length beforehand)?

(I recall @nnethercote replacing quite a few string concatenations with Arrays a few years ago, since it was more efficient, hence my question.)